Alliance.Client/Extensions/VOIP/Behaviors/PBVoiceChatHandlerClient.cs

```diff
@@ -1,1175 +1,1065 @@
+			}
+
+			return -1;
+		}
+
+		public override void OnPlayerDisconnectedFromServer(NetworkCommunicator networkPeer)
+		{
+			base.OnPlayerDisconnectedFromServer(networkPeer);
+			MissionPeer missionPeer = GameNetwork.MyPeer?.GetComponent<MissionPeer>();
+			MissionPeer component = networkPeer.GetComponent<MissionPeer>();
+			if (component?.Team == null || missionPeer?.Team == null)
+			{
+				return;
+			}
+		}
+
+
+		private static float CalculateVolume(Vec3 speakerPosition, Vec3 listenerPosition, float distanceCutoff, float maxVolume)
+		{
+			float distance = speakerPosition.Distance(listenerPosition);
+
+			// Controls the smoothness of the volume change between the 2 positions, higher takes more time for the sound to get louder
+			float smoothnessFactor = 5.0f;
+
+			// Exponential function for smoother increase in volume as listener gets closer
+			float exponentialVolume = smoothnessFactor * (1.0f - (float)Math.Exp(-distance / distanceCutoff));
+
+			// Scale the exponent to maximum volume
+			float volume = maxVolume * MathF.Pow(1.0f - MathF.Clamp(distance / distanceCutoff, 0.0f, 1.0f), exponentialVolume);
+
+			// Limit minimum and maximum volume 
+			float minVolume = MathF.Max(0.0f, volume);
+			volume = MathF.Min(maxVolume, minVolume);
+
+			return volume;
+		}
+
+		public class PlayerVoiceData
+		{
+			private const int PlayDelaySizeInMilliseconds = 150;
+			private const int PlayDelaySizeInBytes = 3600;
+			private const float PlayDelayResetTimeInMilliseconds = 300f;
+
+			public readonly MissionPeer Peer;
+			public readonly Vec3 Position;
+			public readonly Agent Agent;
+
+			private readonly Queue<short> _voiceData;
+			private readonly Queue<short> _voiceToPlayInTick;
+			private int _playDelayRemainingSizeInBytes;
+			private MissionTime _nextPlayDelayResetTime;
+			private bool _isSpeaking;
+
+			public WaveOutEvent waveOut { get; private set; }
+			public BufferedWaveProvider waveProvider { get; private set; }
+			public PanningSampleProvider panProvider { get; private set; }
+			public VolumeSampleProvider volumeProvider { get; private set; }
+
+			public bool IsReadyOnPlatform { get; private set; }
+			private OpusDecoder _decoder;
+
+			public PlayerVoiceData(MissionPeer peer, MixingSampleProvider mixer, Agent agent = null)
+			{
+				Peer = peer;
+				Agent = agent;
+				_voiceData = new Queue<short>();
+				_voiceToPlayInTick = new Queue<short>();
+				_nextPlayDelayResetTime = MissionTime.Now;
+				_decoder = OpusDecoder.Create(SAMPLE_RATE, 1);
+
+				//waveOut = new WaveOutEvent();
+
+				// Initialize NAudio components
+				waveProvider = new BufferedWaveProvider(new WaveFormat(SAMPLE_RATE, 16, 1));
+				waveProvider.BufferLength = VOICE_RECORD_MAX_CHUNK_SIZE_IN_BYTES;
+				//waveOut.Init(waveProvider);
+
+				volumeProvider = new VolumeSampleProvider(waveProvider.ToSampleProvider());
+				panProvider = new PanningSampleProvider(volumeProvider);
+				volumeProvider.Volume = 1;
+
+
+				//waveOut.Init(panProvider);
```

Comment:
Are commented code still useful ?
Yes => Add comment telling why (With TODO maybe?)
No => Remove it